### DISCUSSION TITLE
Use docker hub account's pull rate limit

### DISCUSSION THREAD

**Discussion by masterjuggler** [nextcloud/all-in-one#2667] [Ideas] [answered]
I'm running nextcloud aio on an ubuntu server. I input my credentials following `docker login` both as user and root, and confirmed that manually pulling an image in the host hits my account's rate limit of 200 using the method here: https://docs.docker.com/docker-hub/download-rate-limit/. 

However, when I spin up nextcloud aio, I can see the images pulled are hitting the anonymous rate limit of 100 instead, which can add up quickly when I'm messing with things. Let me know if there's any information I failed to provide that would help.

**Comment by Zoey2936** [comment]
AIO itself pulls as anonymous, but I think the best why for AIO would be to become part of the docker hub verified publischer/oss program, since image of accounts that are inside those programs, have no rate limit to their images by everyone: https://docs.docker.com/docker-hub/download-rate-limit

**Reply by masterjuggler** [reply]
That does sound like a good way to go. Do you know who needs to be pinged that would either be able to get the ball rolling on this, or to tell us why it's not feasible?

And in the mean time, I'd like to know if it's possible to have AIO pull as my logged in account so I'm not depleting the anonymous rate limit for my IP since I'm not the only one on my network using docker.

**Reply by masterjuggler** [reply]
@szaimen The anonymous pull rate limit is really the last thing stopping me from deploying AIO in production. Is there any way to either have it use my account's credentials or for AIO to become part of the verified publisher program?

**Comment by alekslyse** [comment]
I have the same issue and it impossible to install nextcloud. I tried to login by entering the mastercontainer and docker login on www-data and root account, but that didnt help. If I pulled all images before I started the install it didnt fail before I came to the installation (after domain verification), then failed again since it check if the images are up to date and hit the limit again. Something have to be done with that pull limit

**Comment by szaimen** [comment, marked as answer]
Hey, just as heads-up: I created https://github.com/nextcloud/all-in-one/pull/3313 which will reduce the amount of pulls AIO does.

**Reply by szaimen** [reply]
This is now released with v7.2.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel
